addons/point_of_sale/models/res_config_settings.py

```diff
@@ -247,7 +247,7 @@ def _compute_pos_fiscal_positions(self):
     def _compute_pos_tip_product_id(self):
         for res_config in self:
             if res_config.pos_iface_tipproduct:
-                res_config.pos_tip_product_id = res_config.pos_config_id.tip_product_id
+                res_config.pos_tip_product_id = self.env['product.product'].search([('default_code', '=', 'TIPS')])
```

Comment:
If I rember well the default tips product in pos.config model is only set when the model is intalled, if the tip product is created afterwards it won't be set in `res_config.pos_config_id.tip_product_id`

Reply:
we should keep the `res.config.settings` this way. If you want to conditionally put a value in the tip_product_id, do it in `write` method of the `pos.config` model.

Reply:
> we should keep the res.config.settings this way.

Why? Because "res.config.settings" is the "temporary" place where we modify the "pos.config". As much as possible, we still put the business logic in "pos.config". So, I'd suggest to put the logic of setting default tips in the "pos.config" model.